src/griddyn/governors/GovernorSteamNR.cpp

```diff
@@ -75,7 +75,7 @@ void GovernorSteamNR::dynObjectInitializeB (const IOdata & /*inputs*/,
     m_state[offset + 1] = 0;
     m_state[offset + offset] = desiredOutput[0];
 
-    Pset = ((Generator *)getParent ())->getPset ();
+    Pset = dynamic_cast<Generator*> (getParent ())->getPset ();
```

Comment:
this should be static_cast,  it is guaranteed that the parent is a generator by construction if it gets here

Reply:
```
coreObject *GovernorSteamNR::clone (coreObject *obj) const
{
    GovernorSteamNR *gov;
    if (obj == nullptr)
    {
        gov = new GovernorSteamNR ();
    }
    else
    {
        gov = dynamic_cast<GovernorSteamNR *> (obj);
        if (gov == nullptr)
        {
            coreObject::clone (obj);
            return obj;
        }
}
```
should be
```
coreObject *GovernorSteamNR::clone (coreObject *obj) const
{
    GovernorSteamNR *gov;
    if (obj == nullptr)
    {
        gov = new GovernorSteamNR ();
    }
    else
    {
        gov = static_cast<GovernorSteamNR *> (obj);
    }
}
```
then? static_cast<T>(obj) won't be null unless obj is

Reply:
This class is not operational, it will be rewritten at a later time.  The model is wrong, and the code is an old copied placeholder.  I am waiting to get a more generic control system model running, before replacing it though.